gcode/src/lexer.rs

```diff
@@ -253,11 +275,22 @@ mod tests {
 
     #[test]
     fn skip_whitespace() {
-        let mut lexer = Lexer::new("  \n\r\t  ");
+        let mut lexer = Lexer::new("  \r\t  ");
 
         lexer.skip_whitespace();
 
         assert_eq!(lexer.current_position, lexer.src.len());
+        assert_eq!(lexer.current_line, 0);
+    }
+
+    #[test]
+    fn respect_newlines() {
+        let mut lexer = Lexer::new("\n\rM30garbage");
+
+        let token = lexer.tokenize_newline();
+        assert_eq!(token.expect("Failed.").kind, TokenType::Newline);
+
+        assert_eq!(lexer.current_position, 1);
```

Comment:
I normally prefer to write it like this:

```suggestion
        let token = lexer.tokenize_newline().unwrap();
        
        assert_eq!(token.kind, TokenType::Newline);
        assert_eq!(lexer.current_position, 1);
```